### ISSUE TITLE
[Feature]: i18n (again)

### ISSUE THREAD

**Issue by thombruce** [thombruce/tnt-2024#52]
### Feature request

Don't know how I missed this but the Nuxt i18n module **does** in fact have layers support: https://v8.i18n.nuxtjs.org/guide/layers

That isn't the immediate concern, but I wonder if it might fix the error we're facing...

```
 ERROR  [nuxt] [request error] [unhandled] [500] Cannot read properties of undefined (reading 't')
```

The layers approach seems to require lazy loading, which I'm also looking at as a potential solution to this error.

For the time being, I am removing any used translations so that I can release a "fixed" version. I'll then circle back to addressing this.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by thombruce** [comment]
Given that i18n is presently non-existent in TNT... and I really am having a hard time getting it to work as either layer or module...

I am going to remove it almost entirely for now, shelve this and we'll revisit it with a clearer plan in the future.

Remember too that v8 of the Nuxt i18n module isn't yet officially released. _We've been using a release candidate, but there is still a lot on the roadmap's backlog._

It just isn't important enough that we need to work this hard to get it working for now. TNT users, including myself, can install and setup i18n themselves as needed.

For now, we'd rather move onto building with TNT. _Sounds dangerous..._

**Comment by thombruce** [comment]
Possibly relevant to my problems: https://volta.net/nuxt-modules/i18n/issues/2044

Unsure if I do want to opt for the lazy-loading approach, but when I was giving it a try this does reflect my experience somewhat.

Preferred the module approach... with which the only issue was that `t()` was undefined. Meaning it wasn't being imported automatically and so wasn't available to `yarn generate`, thus a failure that went unidentified.

I feel as though that, the module approach, would be easier to fix... and we're still mostly setup to utilise it (I've only removed translations usage from the `Settings.vue` component). Maybe I try to get that imported properly. Maybe I import it explicitly! Maybe then it works fine...

**Comment by thombruce** [comment]
Attempting a workaround whereby I am changing the `Settings.vue` component to `Settings.client.vue` (client-only component).

This successfully builds on `yarn generate`... and I am able to change languages on the `yarn build && yarn preview` site.

_Promising..._

But not ideal if translations are only permitted in client-side components.

**Comment by thombruce** [comment]
Good news: Following the change to client-only settings component, translations are now fully working!

_Possible to create our own `$t()`-like composable that intelligently yields one value on the client and has a fallback on server/generate?_

Alternatives:

1. Create a `I18n.client.vue` component that takes a key as a prop and yields the appropriate translation in place.

_This wouldn't be useable for form labels. Definitely a client-only composable that yields a non-i18n-dependent value if the environment is not client would be preferable._

**Comment by thombruce** [comment]
Some hacking and cobbling together this... also the composable doesn't initially load correctly when used as the label for the 'Language' `<TntSelect>` component, so I've still had to wrap this with `<ClientOnly></ClientOnly>`. But otherwise...

```ts
export const i18n = (key: string, fallback?: string) => {
  if (process.client) {
    const { t } = useI18n()
    return t(key, fallback)
  } else {
    return key
  }
}
```

This composable is useable in a form like `{{ i18n('settings', i18n('tnti18n.settings')) }}` (typical usage will be more like `{{ i18n('settings') }}`) and successfully provides translations in the client environment.

The app generates successfully and I am able to build and preview the behaviour properly.

This is done.

_Super hacky_, but done. I am not going to close this yet, as we will revisit in future but will label it `status: wontfix` for the time being whilst we wait on an official release of the Nuxt i18n module. At that point, we will revisit.

**Comment by thombruce** [comment]
Officially the `v0.11.0` release: 9d3c1600b42a9fd6480d12166c62f6e144742853

_Leaving open for future discussion and revision._
